### ISSUE TITLE
Feature request: Add ability to change speed of animation in CSS3

### ISSUE THREAD

**Issue by patrik-krajcovic2** [w3c/csswg-drafts#8265]
There should be a property that changes animation speed, I don't think “animation-duration”. I want it because it will be useful here: https://github.com/black7375/Firefox-UI-Fix/issues/551

Proposed name for new property: animation-speed

Old issue link: https://github.com/whatwg/html/issues/8656

Specification link: https://w3c.github.io/csswg-drafts/css-animations/#animation

**Comment by Loirooriol** [comment]
I'm not even sure what values this would accept. If you are animating `margin-left`, the speed would be like `calc(1px / 1s)`? And for `opacity`, `calc(1 / 1s)`? And what's the speed if you are interpolating discretely between `width: min-content` to `width: max-content`?

**Comment by patrik-krajcovic2** [comment]
I don't think JavaScript, but only CSS.

**Comment by patrik-krajcovic2** [comment]
And I thought that “animation-speed” would be a new property because it is difficult to change speed of “animation” property in pure CSS. Click to first link which I posted and user called black7375 explains it.

**Comment by cdoublev** [comment]
I am not sure to understand your use case from reading the first link, but you may be interested by [`Animation.playbackRate`](https://developer.mozilla.org/en-US/docs/Web/API/Animation/playbackRate).

#980 is about adding the corresponding CSS property in CSS Animation 2.

> And what's the speed if you are interpolating discretely between `width: min-content` to `width: max-content`?

It could default to 1.

**Comment by patrik-krajcovic2** [comment]
I don't think this: https://developer.mozilla.org/en-US/docs/Web/CSS/animation-duration but similar thing.

**Comment by cdoublev** [comment]
Hmm, sorry, I did not understand. You want `animation-duration` to depend on the animated text length. So a theoretical `animation-speed` would define a dynamic value for it.

But speed is coupled to `<length>` (distance) whereas `animation` can animate an integer, an angle, a resolution, etc... So you would always compute `animation-speed` from the block width of the animated element, I guess, which feels weird when animating properties that do not represent a distance, eg. `opacity` or `color`.

```css
.animated {
  animation-name: animateWithSpeed;
  animation-speed: calc(1px / 1ms);
}
@keyframes animateWithSpeed {
  to {
    opacity: 0;
    color: purple;
    transform: translateX(100%);
  }
}
```

**Comment by patrik-krajcovic2** [comment]
What you thought with these five words “So a theoretical animation-speed”?

**Comment by patrik-krajcovic2** [comment]
I don't think “Animation.playbackRate”, nor “animation-duration”, but new property called “animation-speed” which would calculate the speed of animation of in example text, or image. I will show an example moving text animation for text written in HTML document with attached GIF (And speed for this animation should be controlled with that new property):

![Moving text GIF](https://user-images.githubusercontent.com/69351811/210054313-9524eed5-d418-47ef-be2b-ec0b3420d5d8.gif)

**Comment by patrik-krajcovic2** [comment]
Sorry for German language GIF, but I didn't find a better GIF.

**Comment by patrik-krajcovic2** [comment]
And if you look at the GIF, the moving text have speed, and the speed should be controlled with the proposed “animation-speed” property.

**Comment by patrik-krajcovic2** [comment]
The “animation-duration” property controls animation duration (Which work similar), but not speed and the “Animation.playbackRate” controls animation playback rate, not speed.

**Comment by patrik-krajcovic2** [comment]
And finally the proposed “animation-speed” property would truly control speed.

**Comment by Loirooriol** [comment]
You don't define "speed" in general. I don't think this concept makes sense for most properties. And for the ones that would make sense, it would have different dimensions, so it's not clear how you would specify it.

**Comment by patrik-krajcovic2** [comment]
Just give me a moment I will edit the GIF for better understanding.

**Comment by patrik-krajcovic2** [comment]
Slower:
![Moving text GIF](https://user-images.githubusercontent.com/69351811/210055818-160e968d-350d-4c56-982d-9d1bebce8c18.gif)
Faster:
![Moving text GIF (Faster)](https://user-images.githubusercontent.com/69351811/210055833-0820333d-8962-4535-b532-7086135931da.gif)

**Comment by patrik-krajcovic2** [comment]
Just imagine it that this GIF is HTML text edited by using CSS property “animation-speed”. Just imagine it, please.

**Comment by Loirooriol** [comment]
Editing the gif does not define "speed" in general. Let's say I have this animation with a duration of `100s`:

```css
0% { width: min-content }
10% { width: max-content }
20% { width: 20% }
30% { width: 30% }
40% { width: 40px }
50% { width: 50px }
```

The values first change discretely (NaN speed?), then there is an interpolation from `20%` to `30%` at a rate of `calc(1% / 1s)`, then a interpolation from `30%` to `40px` at a rate of, uh, `calc((4px - 3%) / 1s)`? And it goes on from `40px` to `50px` at a rate of `calc(1px / 1s)`.

Please explain how you would express this in your `animation-speed` property. And this is just for a single property.

**Comment by patrik-krajcovic2** [comment]
In example

animation-speed: speed=10px/1s

Or:

animation-speed: speed=50px/1s (This would be faster than 10px/1s)

**Comment by patrik-krajcovic2** [comment]
Legend:

px = Pixel
s = Second

**Comment by cdoublev** [comment]
I think I do not understand and I do not want to pollute your discussion. 

I reproduced your GIFs in this [CodePen](https://codepen.io/creativewave/pen/MWByXJP?editors=0010). I defined the playback rate ([*Speed control*](https://drafts.csswg.org/web-animations-1/#speed-control)) of the first text via the Web Animation API. I would have used the `animation-playback-rate` CSS property if it existed.

Let me know if I am still off topic or not. =)
